### ISSUE TITLE
Responses from fine-tuned BlenderBot model

### ISSUE THREAD

**Issue by tunde99** [facebookresearch/ParlAI#5025]
**Bug description**
I generated responses from a fine-tuned BlenderBot 90M model using beam decoding and got some unusual results for some of the responses. 

Is there any explanation why the model output is readable in the second episode but displayed lists of single numbers in the first episode, aside from the reported automatic metrics? 

**Reproduction steps**
parlai eval_model -mf [my_model_file] -t jsonfile --jsonfile-datapath [my_data_path] --jsonfile-datatype-extension True --datatype test -bs 16 --skip-generation False --report-filename [my_report_path] --world-logs [my_world_logs] --inference beam --beam-size 10 --beam-context-block-ngram 3 --beam-block-ngram 3 --beam-block-full-context True 

```
{"dialog": [[{"text": "what is the relation between a mill and a power?", "episode_done": true, "id": "data/data_test.jsonl", "eval_labels": [""]}, {"id": "TransformerGenerator", "episode_done": false, "text": "[ 1 ] [ 2 ] [ 3 ] [ 4 ] [ 5 ] [ 6 ] [ 7 ] [ 8 ] [ 9 ] [ 10 ] [ 11 ] [ 12 ] [ 15 ] [ 16 ] [ 17 ] [ 18 ] [ 19 ] [ 21 ] [ 23 ] [ 22 ] [ 24 ] [ 27 ] [ 28 ] [ 33 ] [ 38 ] [ 41 ] [ 58 ] [ 43 ] [ 47 ] [ 46 ] [ 48 ] [ 45 ] [ 60 ] [ 44 ] [ 49 ] [ 50 ] [ 55 ] [ 59 ] [ 40 ] [ 52 ] [ 53 ] [ 54 ] [", "beam_texts": [["[ 1 ] [ 2 ] [ 3 ] [ 4 ] [ 5 ] [ 6 ] [ 7 ] [ 8 ] [ 9 ] [ 10 ] [ 11 ] [ 12 ] [ 15 ] [ 16 ] [ 17 ] [ 18 ] [ 19 ] [ 21 ] [ 23 ] [ 22 ] [ 24 ] [ 27 ] [ 28 ] [ 33 ] [ 38 ] [ 41 ] [ 58 ] [ 43 ] [ 47 ] [ 46 ] [ 48 ] [ 45 ] [ 60 ] [ 44 ] [ 49 ] [ 50 ] [ 55 ] [ 59 ] [ 40 ] [ 52 ] [ 53 ] [ 54 ] [", -17.021757125854492]], "metrics": {"exs": 1, "accuracy": 0.0, "precision": 0.0, "recall": 0.0, "f1": 0.0, "bleu-4": 0.0, "clen": 11.0, "ctrunc": 0.0, "ctrunclen": 0.0, "llen": 1.0, "ltrunc": 0.0, "ltrunclen": 0.0, "loss": 11.76684284210205, "ppl": 128906.52935959311, "token_acc": 0.0, "token_em": 0.0, "gen_n_toks": 129.0}}]], "context": [], "metadata_path": ""}
{"dialog": [[{"text": "what is the relation between a boat and water?", "episode_done": true, "id": "data/data_test.jsonl", "eval_labels": [""]}, {"id": "TransformerGenerator", "episode_done": false, "text": "there is a boat in the water , and a boat on the opposite side of the river .", "beam_texts": [["there is a boat in the water , and a boat on the opposite side of the river .", -11.528935432434082], ["there is a boat on the water , and a boat in the boat .", -11.653779983520508], ["there is a boat in the water , and a boat on the other side of the river .", -11.69117259979248], ["there is a boat on the water , and a boat in the river .", -11.998381614685059], ["there is a boat in the water , but it is not a boat .", -12.112858772277832], ["there is a boat in the water , and a boat on the opposite side of the river , which is situated at the bottom of the lake .", -13.940680503845215], ["there is a boat in the water , and a boat on the opposite side of the river , which is situated on the banks of the canal .", -14.017498016357422], ["there is a boat in the water , and a boat on the other side of the river , and the boat is on the opposite side of it .", -14.248844146728516], ["there is a boat in the water , and a boat on the other side of the river , and the boat is on the opposite side of it , and there is water on both sides .", -15.551864624023438], ["there is a boat in the water , and a boat on the other side of the river , and the boat is on the opposite side of it , and there is water on either side .", -15.681404113769531], ["there is a boat in the water , and a boat on the other side of the river , and the boat is on the opposite side of it , and there is no boat in it .", -15.96416187286377], ["there is a boat in the water , and a boat on the other side of the river , and the boat is on the opposite side of it , and there is no boat in sight .", -15.973238945007324]], "metrics": {"exs": 1, "accuracy": 0.0, "precision": 0.0, "recall": 0.0, "f1": 0.0, "bleu-4": 0.0, "clen": 10.0, "ctrunc": 0.0, "ctrunclen": 0.0, "llen": 1.0, "ltrunc": 0.0, "ltrunclen": 0.0, "loss": 11.354619026184082, "ppl": 85358.81693639157, "token_acc": 0.0, "token_em": 0.0, "gen_n_toks": 21.0}}]], "context": [], "metadata_path": ""}
```

**Additional context**
I did not experience this issue when I evaluated using BlenderBot 90M model from the zoo.

**Comment by klshuster** [comment]
This may be an issue with the data you've used to fine-tune, I would double check that all of your texts and labels are appropriate

Could you also share your fine-tuning command?

**Comment by tunde99** [comment]
> This may be an issue with the data you've used to fine-tune, I would double check that all of your texts and labels are appropriate
> 
> Could you also share your fine-tuning command?

parlai train_model \
--task blended_skill_talk,wizard_of_wikipedia,convai2:normalized,jsonfile \
--jsonfile-datapath [my_datapath] \
--jsonfile-datatype-extension True \
--multitask-weights 1,3,3,1 \
-m transformer/generator \
--init-model zoo:tutorial_transformer_generator/model \
--dict-file zoo:tutorial_transformer_generator/model.dict \
--embedding-size 512 \
--n-layers 8 -\
-ffn-size 2048 \
--dropout 0.1 \
--n-heads 16 \
--learn-positional-embeddings True \
--n-positions 512 \
--variant xlm \
--activation gelu \
--fp16 True \
--text-truncate 512 \
--label-truncate 128 \
--dict-tokenizer bpe \
--dict-lower True \
-lr 1e-06 \
--optimizer adamax \
--lr-scheduler reduceonplateau \
--gradient-clip 0.1 \
-veps 0.25 \
--betas 0.9,0.999 \
--update-freq 1 \
--attention-dropout 0.0 \
--relu-dropout 0.0 \
--skip-generation True \
-vp 15 \
-stim 60 \
-vme 20000 \
-bs 16 \
-vmt ppl \
-vmm min \
--save-after-valid True \
--model-file [my_model_file] \
--add-missing-turns all

**Comment by klshuster** [comment]
command looks good to me. probably an issue with the fine-tune data maybe? could you try training one without your custom task and seeing if it works properly?

**Comment by tunde99** [comment]
@klshuster I tried generating responses with another set of test examples and I did not experience the issue. I will proceed to close this thread. Thank you for your responses.
